app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.java

```diff
@@ -305,22 +306,32 @@ private void updateCurrent() {
                             if (!statuses.isEmpty()) {
                                 filterStatuses(statuses);
 
-                                // Working around a bug when Mastodon API doesn't return the first
-                                // status because of string "id < maxId". Hacking with ID doesn't
-                                // help.
-                                if (!this.statuses.isEmpty()) {
-                                    Either<Placeholder, Status> firstOld = this.statuses.get(0);
-                                    this.statuses.clear();
-                                    this.statuses.add(firstOld);
-                                } else {
-                                    this.statuses.clear();
+
+                                if (!this.statuses.isEmpty() && topId != null) {
+                                    // clear old cached statuses
+                                    Iterator<Either<Placeholder, Status>> iterator = statuses.iterator();
+                                    while (iterator.hasNext()) {
+                                        Either<Placeholder, Status> item = iterator.next();
+                                        if(item.isRight()) {
```

Comment:
You could shorten it if you put id into the variable and left only one comparison

Reply:
Could also find the first newer one and delete everything older but not sure if there's any performance/readability benefit